api/bonsai_api/crud/sample.py

```diff
@@ -206,10 +207,9 @@ async def get_samples_summary(
 ) -> List[SampleSummary]:
     """Get a summay of several samples."""
     # build query pipeline
-    pipeline = []
-    if include_samples is not None and len(include_samples) > 0:
+    pipeline: list[dict[str, Any]] = []
+    if isinstance(include_samples, list) and len(include_samples) > 0:
```

Comment:
Hmm. What do you want to test here? Based on the types specified the previous comparison seems cleaner and more accurate. I don't think `isinstance` is the best tool here.

If you see unexpected behaviour here it might be worth double checking the actual values in the code. I suspect something is not behaving as expected here.

Anyway, just my thoughts. Maybe not worth sweating over if already merged and working. But I think worth taking a closer look before adding more `isinstance`.